### ISSUE TITLE
AbstractResource must be compatible with ResourceInterface::populateFromResponse

### ISSUE THREAD

**Issue by ProPheT777** [php-opencloud/openstack#181]
Hello, 

Since migration to php 7.2 I got this error, with the latest tagged version : 

```
Compile Error: Declaration of OpenStack\Common\Resource\AbstractResource::populateFromResponse(Psr\Http\Message\ResponseInterface $response): OpenStack\Common\Resource\AbstractResource must be compatible with OpenStack\Common\Resource\ResourceInterface::populateFromResponse(Psr\Http\Message\ResponseInterface $response): OpenStack\Common\Resource\ResourceInterface

  at vendor/php-opencloud/openstack/src/Common/Resource/AbstractResource.php:16
```

**Comment by haphan** [comment]
SDK has not been tested with 7.2 since it is relatively new.

I will add php 7.2 to travis.

**Comment by ProPheT777** [comment]
Ok, nice to Know, thanks you, I'll sens PR if I have free Time to fixe that

**Comment by haphan** [comment]
@ProPheT777 I just ran tests with php 7.2 and unfortunately, the SDK may require a bit of refactoring as well as breaking backward compatibility.
```
Fatal error: Cannot use OpenStack\ObjectStore\v1\Models\Object as Object because 'Object' is a special class name in /work/tests/unit/ObjectStore/v1/Models/ContainerTest.php on line 11```

**Comment by phansys** [comment]
This fix has nothing to do with PHP 7.2, since the same result occurs with PHP 7.0 or 7.1: https://3v4l.org/dkLdi.

**Comment by haphan** [comment]
Fix issue should be fixed in latest 3.x and 2.x version. Let me know if it is not the case.
